### PR TITLE
wip: Add create-ultra-app to repo

### PR DESCRIPTION
**Pull request by B3nten**
- [ ] Test template functionality
- [x] Enhance CLI text
- [ ] Create imported files locally

### CODE REVIEW

**Review by deckchairlabs — COMMENTED** [review]
Damn dude, this looks so nice. Well done!

**Comment by deckchairlabs** [comment]
Whats the status of this, is it ready to merge?

**Comment by deckchairlabs** [comment]
One tiny request, can we get some color on the questions asked? Maybe the same blue used in the "congrats" message at the end

**Comment by B3nten** [comment]
Will need to update the imports to link to the files in this repo, and some of the templates need to be tested to make sure they work.

**Comment by cdoremus** [comment]
Great job Benton! 🚀🚀🚀

**Comment by deckchairlabs** [comment]
> I think we should, at least for now, go the route of creating all required files locally.

Is this in reference to the files that are downloaded from the repo?

**Comment by benton-droplab** [comment]
> > I think we should, at least for now, go the route of creating all required files locally.
> 
> Is this in reference to the files that are downloaded from the repo?

The files in `modules` that are imported rather than created. For example, `useFlushAsyncWhateverTheNameIs`

**Comment by deckchairlabs** [comment]
Oh no, right I see what you mean. Yeah I'd like the files to be local, gives developers a chance to modify if they so choose.

**Comment by B3nten** [comment]
> Oh no, right I see what you mean. Yeah I'd like the files to be local, gives developers a chance to modify if they so choose.

Should be a trivial change, I can push something up tonight.

For the Twind issue that was resolved, was there something in particular that was implemented wrong?

**Comment by deckchairlabs** [comment]
> > Oh no, right I see what you mean. Yeah I'd like the files to be local, gives developers a chance to modify if they so choose.
> 
> Should be a trivial change, I can push something up tonight.
> 
> For the Twind issue that was resolved, was there something in particular that was implemented wrong?

@mashaal best to answer that

**Comment by mashaal** [comment]
> > Oh no, right I see what you mean. Yeah I'd like the files to be local, gives developers a chance to modify if they so choose.
> 
> Should be a trivial change, I can push something up tonight.
> 
> For the Twind issue that was resolved, was there something in particular that was implemented wrong?

It seems it was missing the twind.ts Deno block here. https://github.com/exhibitionist-digital/ultra/blob/main/examples/with-twind/twind.ts#L4-L12

And then I got a useDehydratedState issue with react-query. https://github.com/exhibitionist-digital/ultra/blob/main/examples/with-react-query/src/hooks/useDehydrateReactQuery.tsx

That issue raised covered a heap of surface area. react-query, react-router, trpc, twind, helmet -- I was able to get it to work after a few mins of localising the twind and query dependencies, but I'm certain this level of integration could be a bit more elegant. https://github.com/exhibitionist-digital/ultra/issues/153

**Comment by B3nten** [comment]
> > > Oh no, right I see what you mean. Yeah I'd like the files to be local, gives developers a chance to modify if they so choose.
> > 
> > 
> > Should be a trivial change, I can push something up tonight.
> > For the Twind issue that was resolved, was there something in particular that was implemented wrong?
> 
> It seems it was missing the twind.ts Deno block here. https://github.com/exhibitionist-digital/ultra/blob/main/examples/with-twind/twind.ts#L4-L12
> 
> And then I got a useDehydratedState issue with react-query. https://github.com/exhibitionist-digital/ultra/blob/main/examples/with-react-query/src/hooks/useDehydrateReactQuery.tsx
> 
> That issue raised covered a heap of surface area. react-query, react-router, trpc, twind, helmet -- I was able to get it to work after a few mins of localising the twind and query dependencies, but I'm certain this level of integration could be a bit more elegant. #153

sheet should have been provided here https://github.com/B3nten/ultra/blob/create/lib/create/modules/twind/content/twind.ts

But regardless it was missing useDehydrateReactQuery which was obviously an issue. I'll try to get something up tonight that should fix those issues. And try to make it more elegant

**Comment by B3nten** [comment]
Okay, I think this is ready to merge. The current `create` script is broken so this wouldn't be any worse. I've disabled TRPC until I can confirm it works.

File: create.ts [hunk 1]
```diff
@@ -1,401 +1,104 @@
...
-
-  const context: TaskContext = {
+// IMPORTS
+import { Config, Libraries } from "./lib/create/common/config.ts";
+import { ask, confirm } from "./lib/create/common/ask.ts";
+import { createUltraApp } from "./lib/create/common/createUltraApp.ts";
+import { c, gradient } from "./lib/create/common/styling.ts";
+
+//INIT
+console.log(
```
Comment (deckchairlabs) [inline create.ts:8]: whats this for? seems odd

Comment (B3nten) [reply create.ts:8]: Both functions are for colourizing text. `c(increment, text)` takes an increment (num) and adjusts the hue accordingly, it's for having each option be a different colour. `gradient()` is essentially the same function used in the Ultra CLI.

File: create.ts [hunk 2]
```diff
@@ -1,401 +1,104 @@
...
+  `\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n`,
```
Comment (deckchairlabs) [inline create.ts:9]: Sorry, highlighted wrong line!

Comment (B3nten) [reply create.ts:9]: It's just a bunch of newlines to clear the terminal. Thought it looked cleaner.

Comment (deckchairlabs) [reply create.ts:9]: I thought you'd just be able to use `console.clear()` to do that

Comment (B3nten) [reply create.ts:9]: That works in the browser but not in system consoles. There might be a deno method to do so, or something like `console.log("\033[2J\033[1;1H")`

Comment (deckchairlabs) [reply create.ts:9]: Ya joking, well, do something like this I guess. https://github.com/UltiRequiem/deno-clear/blob/main/mod.ts

**Pull request closed without merging**